utils/dump_leaderboard.go

```diff
@@ -141,3 +173,56 @@ func main() {
 		}
 	}
 }
+
+func dailyRanks(results [25]map[int][]Result, members []Member) {
+	// Sort the results for each day.
+	for day := range results {
+		sort.Sort(ByTimestamp(results[day][1]))
+		sort.Sort(ByTimestamp(results[day][2]))
+	}
+
+	// Assign rank to members for each day.
+	for day := range results {
+		for ndx, member := range members {
+			members[ndx].Ranks[day][0] = getRank(results[day][1], member.Name)
+			members[ndx].Ranks[day][1] = getRank(results[day][2], member.Name)
+		}
+	}
```

Comment:
I'd rather have the members array be frozen once we leave main and enter into what feels like a dumping routine.

What would you think about making a separate/parallel ranks array of the same dimension as members, given that Ranks is just a helper used by this dumping mode? So something like this:

```go
ranks := make([][26][3]string, len(members))
for day := range results {
  for i, member := range members {
    ranks[i][day][1] = getRank(results[day][1], member.Name)
    ranks[i][day][2] = getRank(results[day][2], memberName)
  }
}
```

Then the loop below on 198 looks like:

```go
for i := range members {
  member := &member[i]
  rank := &ranks[i]

  ...
}
```

In the first snippet I changed the star indexing in ranks to be 1 and 2 like in the results array. As proposed, line 187 goes from a 1-based star index to a 0-based star index.